docs/arch_develop.md

```diff
@@ -72,6 +72,39 @@ Some components of the core library depend on the `package.json`s of the extensi
     -   Does not restore, it is a superset of what exists in `packages/core` for `configuration.properties`.
     -   To develop for the Amazon Q extension: add all changes to `packages/amazonq/package.json`, EXCEPT for settings that are references by code in the core library, or settings that already exist in the core `package.json`
 
+## Shared vs Common names
+
+In this repo, the keywords **"shared"** and **"common"** have specific meanings in the context of file/folder names.
+
+### "common"
+
+Code within a folder/file that has the "common" keyword implies that it can run in any environment. Examples of environments are: Web mode, or Node.js (local desktop)
```

Comment:
node-only or web-only code would live in a specific place (`.../node/...` or `.../web/...`), right? So everything else can be assumed to be "common". Thus we may not need the "common" name at all.

The tricky part is the transitional phase. Currently we don't have a clear dilineation between anything. So names like `extensionCommon.ts` are an attempt to bridge that transition. But because 80% of all code should be "common", adding "common" everywhere seems much more noisy than adding "node" or "web" to specific parts that are intended to be platform-specific.

Reply:
Mentioned below:
> If there is no "shared" or "common" naming used for the file/folder, then assume it only works in Node.js.

I think it is easiest to consider everything as Node.js specific in this transitional phase. Then as we verify code is `common` we can move that code in to the appropriate `common` folder. This will allow us to defer verifying all our code is common, and instead do it incrementally.

So for now this would essentially look like:
- create a `common` subfolder in the folder you are working in and any existing or new code that you know is `common` should be moved in to there as part of your PR
- Anything outside the `common` subfolder is assumed Node.js only

---

Eventually most of our code will be `common`, and by then we can update the structure to be the reverse (anything not defined is assumed as `common`).

Reply:
> This will allow us to defer verifying all our code is common, and instead do it incrementally.

But that implies we would (eventually) move, say 80% of all code a "common" folder. That's a big lift.

> Eventually most of our code will be common, and by then we can update the structure to be the reverse

And that would again be a big (inverse) lift.

It's a much smaller lift to instead identify things that are truly node-only or web-only. This means we assume that everything is "universal" until proven otherwise. As we discover non-universal pockets, we move only those particular parts to "node" or "web" folders.